### PR TITLE
Rp

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: vividus-agent-reportportal/src/main/java/org/vividus/reportportal/jbehave/AdaptedDelegatingReportPortalStoryReporter.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    @Subscribe
+    public void onAssertionFailure(@SuppressWarnings("unused") AssertionFailedEvent event)
+    {
+        reporter.getLastStep().ifPresent(s -> {
+            failedSteps.add(s);
+            ReportPortal.emitLog(s.getItemId(), itemUuid -> {
+                SaveLogRQ rq = new SaveLogRQ();
+                rq.setItemUuid(itemUuid);
+                rq.setLevel(LogLevel.ERROR.name());
+                rq.setLogTime(Calendar.getInstance().getTime());
```
Comment (valfirst) [inline vividus-agent-reportportal/src/main/java/org/vividus/reportportal/jbehave/AdaptedDelegatingReportPortalStoryReporter.java:96]:
```suggestion
                rq.setLogTime(new Date());
```
?

File: vividus-agent-reportportal/src/main/resources/properties/defaults/default.properties [hunk 2]
```diff
@@ -1,3 +1,4 @@
 bdd.generate-view-after-batches=#{T(String).valueOf(${system.rp.enable} || (T(System).getProperty('rp.enable')?:false))}
```
Comment (valfirst) [inline vividus-agent-reportportal/src/main/resources/properties/defaults/default.properties:1]: is it needed?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-agent-reportportal/build.gradle [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+    implementation project(':vividus-engine')
     implementation project(':vividus-soft-assert')
     implementation project(':vividus-reporter')
     implementation(group: 'com.google.guava', name: 'guava', version: versions.guava)
     implementation(group: 'org.springframework', name: 'spring-context', version: versions.spring)
 
     // ReportPortal
-    implementation(group: 'com.epam.reportportal', name: 'logger-java-log4j', version: '4.0.2')
-    implementation(group: 'com.github.reportportal', name: 'agent-java-jbehave', version: '4.2.1') {
+    implementation(group: 'com.epam.reportportal', name: 'logger-java-log4j', version: '5.1.1')
```
Comment (valfirst) [inline vividus-agent-reportportal/build.gradle:11]:
```suggestion
    implementation(group: 'com.epam.reportportal', name: 'logger-java-log4j', version: '5.1.2')
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-agent-reportportal/src/main/java/org/vividus/reportportal/jbehave/AdaptedDelegatingReportPortalStoryReporter.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.reportportal.jbehave;
+
+import java.util.ArrayList;
+import java.util.Calendar;
```
Comment (valfirst) [inline vividus-agent-reportportal/src/main/java/org/vividus/reportportal/jbehave/AdaptedDelegatingReportPortalStoryReporter.java:20]:
```suggestion
```

File: vividus-agent-reportportal/build.gradle [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
+    implementation project(':vividus-engine')
     implementation project(':vividus-soft-assert')
     implementation project(':vividus-reporter')
     implementation(group: 'com.google.guava', name: 'guava', version: versions.guava)
     implementation(group: 'org.springframework', name: 'spring-context', version: versions.spring)
 
     // ReportPortal
-    implementation(group: 'com.epam.reportportal', name: 'logger-java-log4j', version: '4.0.2')
-    implementation(group: 'com.github.reportportal', name: 'agent-java-jbehave', version: '4.2.1') {
+    implementation(group: 'com.epam.reportportal', name: 'logger-java-log4j', version: '5.1.2')
```
Comment (valfirst) [inline vividus-agent-reportportal/build.gradle:11]:
```suggestion
    implementation(group: 'com.epam.reportportal', name: 'logger-java-log4j', version: '5.1.3')
```
welcome to log4jhell

**Review by valfirst — APPROVED** [review]

File: vividus-agent-reportportal/src/main/java/org/vividus/reportportal/jbehave/AdaptedDelegatingReportPortalStoryReporter.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Override
+    public void scenarioExcluded(Scenario scenario, String filter)
+    {
+        reporter.scenarioNotAllowed(scenario, filter);
+    }
+
+    @Subscribe
+    public void onAssertionFailure(@SuppressWarnings("unused") AssertionFailedEvent event)
```
Comment (uarlouski) [inline vividus-agent-reportportal/src/main/java/org/vividus/reportportal/jbehave/AdaptedDelegatingReportPortalStoryReporter.java:88]: used at 97

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**